persistence/persistence-mybatis/src/main/resources/org/cbioportal/persistence/mybatis/ClinicalDataMapper.xml

```diff
@@ -53,12 +53,24 @@
                 cancer_study.CANCER_STUDY_IDENTIFIER = #{studyIds[0]}
             </if>
             <if test="sampleIds != null">
-                <foreach index="i" collection="sampleIds" open="(" separator="OR" close=")">
-                    cancer_study.CANCER_STUDY_IDENTIFIER = #{studyIds[${i}]} AND sample.STABLE_ID = #{sampleIds[${i}]}
-                </foreach>
+                <if test="studyIds.size() == 1">
+                    cancer_study.CANCER_STUDY_IDENTIFIER = #{studyIds[0]} AND
+                    sample.STABLE_ID IN
+                    <foreach item="item" collection="sampleIds" open="(" separator="," close=")">
+                        #{item}
+                    </foreach>
+                </if>
+                <if test="studyIds.size() > 1">
+                    <foreach index="i" collection="sampleIds" open="(" separator="OR" close=")">
```

Comment:
The reason I didn't do it is our in-memory unit test database H2 doesn't support it, because it's not part of standard SQL, so we can't write unit tests for this case if we use this syntax. And I also didn't want to use MySQL(or any other DBMS) specific features. But of course we can use it if we have to (for virtual cohort for example), but I fixed the problem for now with a new endpoint, which was already needed anyway.